### PR TITLE
feat(learn): add tests to use a callback function in the sort method

### LINKED ISSUE #40724: Change the array used in test or add another case
**Describe your problem and how to reproduce it:**
Below mentioned challenge can be passed using simple .sort() function instead of using .sort((a,b) => a-b)

**Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/functional-programming/return-a-sorted-array-without-changing-the-original-array

**Possible solutions**
1. Change the existing array from [5, 6, 3, 2, 9] to something like [1, 30, 4, 21, 100000]
2. Add another test case with array [1, 30, 4, 21, 100000]

So that the challenge cannot be passed without using .sort((a,b) => a-b) function

### PR DESCRIPTION
**Pull request by Ismailtlem**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

2 tests cases were added and a sentence mentionning that the callback method must be used is added to the challenge description

Closes #40724

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/return-a-sorted-array-without-changing-the-original-array.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ A side effect of the `sort` method is that it changes the order of the elements
 
 # --instructions--
 
-Use the `sort` method in the `nonMutatingSort` function to sort the elements of an array in ascending order. The function should return a new array, and not mutate the `globalArray` variable.
+Use the `sort` method in the `nonMutatingSort` function to sort the elements of an array in ascending order. The function should return a new array, and not mutate the `globalArray` variable. A callback function must be provided to the `sort` method in order for the tests to pass.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/return-a-sorted-array-without-changing-the-original-array.md:15]: I'm not sure the additional note is necessary. It reads to me like we are testing that the camper passes a callback function to sort, when instead we are testing that the function returns the correct value.

**Review by naomi-lgbt — DISMISSED** [review]
Looks good to me.

Tested locally - the following code passes in production but fails these additional tests:
```js
var globalArray = [5, 6, 3, 2, 9];
function nonMutatingSort(arr) {
  return arr.map(el => el).sort()
}
nonMutatingSort(globalArray);
```

I'm not convinced about the addition of the note, but I don't feel strongly enough either way to block the approval. 🙂

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/return-a-sorted-array-without-changing-the-original-array.md (hunk 1 above)
Comment (Ismailtlem) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/return-a-sorted-array-without-changing-the-original-array.md]: I think you are right (y)

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

**Pull request merged**